job.c

```diff
@@ -276,6 +277,60 @@ sig_dfl(void)
     signal(SIGPIPE, SIG_DFL);
 }
 
+#define DQUOTE '\"'
+#define BSLASH '\\'
+#define SPECIALS "()<>[].,:;@"
+#define MAIL_LINE_LEN 998             /* RFC5322 */
+
+char *
+make_mailbox(char *displayname, char *mail_from, char *hostname)
+    /* Produce a "mailbox" header as per RFC5322 sec. 3.2.3
+     * <https://datatracker.ietf.org/doc/html/rfc5322#section-3.2.3> */
+{
+    char specials[sizeof(SPECIALS) + 2];
+    char need_quotes = 0, need_anglebrackets = 0;
+    char c = '\0';
+    char *bpos = NULL, *dpos = NULL;
+    char *buf1 = NULL, *buf2 = NULL;
+
+    snprintf(specials, sizeof(specials), "%s%c%c", SPECIALS, DQUOTE, BSLASH);
+
+    /* This should actually be shorter 'cause the header name is prepended
+     * downstream */
+    buf1 = (char *)calloc(MAIL_LINE_LEN+1, sizeof(char));
+    buf2 = (char *)calloc(MAIL_LINE_LEN+1, sizeof(char));
+
+    /* walk the displayname and rebuild it in buf1 */
+    dpos = displayname;
+    bpos = buf1;
+    while (*dpos) {
+        c = *dpos++;
+        if (strchr(specials, c)) {
+            /* insert escape */
+            if (c == DQUOTE) *bpos++ = BSLASH;
+            need_quotes = 1;
+        }
+        *bpos++ = c;
+    }
+
+    need_anglebrackets = strlen(buf1) > 0;
+
+    if (need_quotes)
+        snprintf(buf2, MAIL_LINE_LEN, "\"%s\"", buf1);
+    else
+        buf2 = strncpy(buf2, buf1, MAIL_LINE_LEN);
+
```

Comment:
everything above here only depends on `displayname`. Consider splitting that function in two, and run the first bit only once instead of doing it every time we send an email?

Reply:
Makes sense. I'd mangle displayname alone at `[fcronconf.c:195](https://github.com/sphakka/fcron/blob/4b7a4b55fefb68ee735eee3cea945156a97e50e3/fcronconf.c#L195)`. Should this new function reside in `fcronconf.c`?